wallet/src/de/schildbach/wallet/ui/AbstractPINDialogFragment.java

```diff
@@ -0,0 +1,130 @@
+package de.schildbach.wallet.ui;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Process;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.bitcoinj.wallet.Wallet;
+
+import de.schildbach.wallet.WalletApplication;
+import de.schildbach.wallet.ui.preference.PinRetryController;
+import de.schildbach.wallet.util.KeyboardUtil;
+import de.schildbach.wallet_test.R;
+
+/**
+ * Created by Hash Engineering on 4/8/2018.
+ */
+
+public abstract class AbstractPINDialogFragment extends DialogFragment {
+
+    protected DialogInterface.OnDismissListener onDismissListener;
+
+    protected AbstractWalletActivity activity;
+    protected WalletApplication application;
+    protected Handler backgroundHandler;
+    protected Wallet wallet;
+    protected PinRetryController pinRetryController;
+
+    protected EditText pinView;
+    protected TextView badPinView;
+    protected Button unlockButton;
+    protected TextView messageTextView;
+
+    protected int dialogFragmentId;
+    protected int dialogTitle;
+
+    AbstractPINDialogFragment()
+    {
```

Comment:
From what I see throughout the code, the opening bracket is usually placed in the same line of method name. Can we do the same here?

Reply:
Yes.